Removes SEO plugin

sgiehl opened the pull request:
### Description:

Matomo was providing a SEO widget, which allowed to fetch some stats around indexed pages on e.g. Google, Bing or Alexa. This widget failed to fetch its stats (on cloud) more and more, making the widget mostly useless.

Also when it comes to “SEO” in a dashboard, this widget under-delivers on what people would hope to see. We therefore decided to remove the widget.

If anyone would like to keep the widget please create an issue for that, and we might consider releasing it as a separate plugin instead.

refs #21712 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed



### Checklist
- [✔] I have understood, reviewed, and tested all AI outputs before use
- [✔] All AI instructions respect security, IP, and privacy rules

sgiehl committed:
Removes SEO plugin (widget)

sgiehl committed:
Removes SEO plugin (widget)

sgiehl committed:
remove remaining ocurrences of SEO plugin and adjust tests

sgiehl committed:
remove remaining ocurrences of SEO plugin and adjust tests

sgiehl committed:
Add update script to ensure SEO plugin is removed on update

sgiehl committed:
Add update script to ensure SEO plugin is removed on update

sgiehl committed:
update version

tsteur commented:
As this is a breaking change, can we add the changelog entry? Wondering actually if we should keep the plugin itself and keep the API returning an empty datatable and only remove the widget so we're not breaking the HTTP API? In Matomo 6 we could then break the API and remove the entire plugin.

sgiehl commented:
@tsteur please discuss this with @mattab. The target was to remove this feature with lowest effort.
Personally I would have kept the feature and only disabled it on cloud (as it seems too much effort to test a possible fix there). Or alternatively move it to a plugin on marketplace, so at least on prem users still finding it valuable could still use it.

tsteur commented:
@sgiehl I think the issue was maybe to remove the widget, not the plugin? 

> The target was to remove this feature with lowest effort.

I'd say it was about the lowest effort while still keeping our promises such as that we won't break our APIs. It could be even fine to only remove the Widget file itself and the rest in Matomo 6. This way it would have been very little effort to just remove the widget file without breaking the HTTP API etc.

> Personally I would have kept the feature and only disabled it on cloud

It's overall fine to remove it as it provides very little value nowadays. 

Breaking the API can be an issue. For example the Matomo Connect WordPress plugin uses the API in https://github.com/braekling/WP-Matomo/blob/master/classes/WP_Piwik/Widget/Seo.php#L14 etc.

sgiehl commented:
In that case, we will move removing that plugin to Matomo 6. If it is causing problems on Cloud we can simply disable the plugin there. On premise it was working as expected anyway.

sgiehl committed:
Fix PR check failures after SEO plugin removal

- Remove stale SEO ignoreErrors entries from PHPStan baseline
- Drop useless @param annotation in the 6.0.0-b1 update (PHPCS)
- Update UI expected screenshots to reflect the removed SEO plugin/widget
  (plugins admin listing, widgetized dashboard, API listing, diagnostics
  config file, marketplace manage-plugins page)

sgiehl committed:
Revert incorrect WidgetizedDashboard screenshots

The previous commit updated these two screenshots to a full-dashboard
render, but that output was produced by a flaky copy-dashboard timeout
cascade in CI, not intended behavior. The widgetized dashboard must show
a single widget, so restore the original expected screenshots.

sgiehl committed:
Update dashboard2 and admin_home screenshots for reduced plugin count

Removing the SEO plugin lowers the activated-plugin count shown in the
CoreHome quick-links widget (69 to 68), which these two screenshots
capture.

chippison commented on core/Updates/6.0.0-b1.php:
```diff
@@ -0,0 +1,51 @@
+/**
+ * Update for version 6.0.0-b1
+ */
+class Updates_6_0_0_b1 extends PiwikUpdates
```
Just wanting to note that we add a core update for this, and `Version.php` is already at 6.0.0-b1. 
Those currently running on that build will miss this upgrade.

sgiehl replied:
I know. But removing the plugin in a git checkout will only remove the config file records, so I didn't consider it needed to update the version. We did not release b1 yet anyway.

chippison reviewed (commented):
There is also a left over reference for `SEO` plugin, but it is on one of our developer console commands and is only run manually. It is on `ComputeAssetSize.php line 109` and will cause an error if the command: `development:compute-js-asset-size`. I have not used this command personally, so it might not even be used at all.

Another thing to note, is if we need to add something in CHANGELOG.md for this removal?

sgiehl committed:
Remove SEO from expected plugins in compute-js-asset-size command

sgiehl committed:
Add changelog entry for SEO plugin removal
